calculate_coverages.py

```diff
@@ -17,21 +21,38 @@ def calculate_coverages(input, output, database):
     ###################################
     gotu_dict = defaultdict(set)
     file_list = glob(input + "/*.sam")
+    file_list_gz = glob(input + "/*.sam.gz")
+    file_list_xz = glob(input + "/*.sam.xz")
+
+    file_list = file_list + file_list_gz + file_list_xz
     for samfile in file_list:
-        with open(samfile.strip(), 'r') as open_sam_file:
+        open_sam_file = None
+        if samfile.endswith(".sam"):
+            open_sam_file = open(samfile.strip(), 'r')
+        elif samfile.endswith(".sam.gz"):
+            open_sam_file = gzip.open(
+                samfile.strip(),
+                mode='rt',
+                encoding='utf-8')
+        elif samfile.endswith(".sam.xz"):
+            open_sam_file = lzma.open(
+                samfile.strip(),
+                mode='rt',
+                encoding='utf-8')
+
```

Comment:
...shouldn't happen but:

```suggestion
else:
    raise IOError("Unrecognized file extension on '%s'." % samfile)
```